mlrun/__init__.py

```diff
@@ -61,6 +61,18 @@ def get_version():
     mlconf.remote_host = mlconf.remote_host or igz_domain
 
 
+_data_stores = None
+
+
+def get_data_stores(secrets=None, override=False):
+    """singleton mlrun data store client"""
+
+    global _data_stores
+    if not _data_stores or override:
+        _data_stores = store_manager.set(secrets)
```

Comment:
`.set()` returning `self`, i.e. a `store_manager` why you're calling it `_data_stores` ?
shouldn't it simply be `_store_manager` ?

Reply:
ok, missed that, i see most of the code is not using the singleton, so i will update it as well (same as removing the .connect() for the db)

Reply:
the issue is that `secrets ` and `db` are not passed to it, so i will add `get(..)` method to the current singleton

Reply:
fixed, use the cached store_manager everywhere